security-analysis/security-analysis-api/src/main/java/com/powsybl/security/action/AbstractTapChangerRegulationAction.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.security.action;
+
+import com.powsybl.iidm.network.ThreeWindingsTransformer;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * An action modifying the regulation of a two or three windings transformer
+ *
+ * @author Etienne Lesot <etienne.lesot@rte-france.com>
+ * @author Anne Tilloy <anne.tilloy@rte-france.com>
+ */
+public abstract class AbstractTapChangerRegulationAction extends AbstractAction {
+
+    private final String transformerId;
```

Comment:
Aybe we could add another intermediate abstract to share transformerId and side ?